Addition of Job scheduler APIs documentation - Jobs API - Locks API (#10663)
opensearch-project/documentation-website: _monitoring-your-cluster/job-scheduler/jobs.md

        "expression": "15 2 1,15 * 1",
        "timezone": "Africa/Abidjan",
        "delay": "none"
      },
      "lock_duration": "no_lock",
      "jitter": "none"
    },
    {
      "job_type": "scheduler_sample_extension",
      "job_id": "jobid1",
      "index_name": ".scheduler_sample_extension",
      "name": "sample-job-it",
      "descheduled": false,
      "enabled": true,
      "enabled_time": "1970-07-23T00:27:45.353Z",
      "last_update_time": "1970-07-23T00:27:45.353Z",
      "last_execution_time": "2025-08-01T22:28:45.357484385Z",
      "last_expected_execution_time": "2025-08-01T22:28:45.353111804Z",
      "next_expected_execution_time": "2025-08-01T22:29:45.353111804Z",
      "schedule": {
        "type": "interval",
        "start_time": "1970-07-23T00:27:45.353Z",
        "interval": 1,
        "unit": "Minutes",
        "delay": "none"
      },
      "lock_duration": 10,
      "jitter": "none"
    }
  ],
  "failures": [],
  "total_jobs": 2
}
```

## Response body fields

The following table lists all response body fields.

| Field | Data type | Description |
| :--- | :--- | :--- |
| `jobs` | Array | Contains all the jobs reported by the Job Scheduler. |

> | `job_type` | String | Shows the plugin that Scheduled the job. |

Author: Fixed

| `job_id` | String | Unique identifier for the job. |
| `index_name` | String | Identifies the index where the job information is stored. |
| `name` | String | Name of the job. Not required to be unique. |
| `descheduled` | Boolean | False if the job is scheduled to be executed. True otherwise. |

> | `enabled` | Boolean | True if the job is scheduled to be executed. False otherwise. |

Author: Enabled is a field that comes from the plugin using Job Scheduler. Descheduled shows whether or not the job is currently scheduled to be ran by Job Scheduler.

| `enabled_time` | String | Shows the time that the job was originally scheduled. |
| `last_update_time` | String | Time stamp of the last time the job was updated. |
| `last_execution_time` | String | Time Stamp of actual time of the last execution. |
| `last_expected_exection_time` | String | Time stamp of expected time of the last execution. |
| `next_expected_execution_time` | String | Time stamp of next scheduled execution time. |

> | `schedule` | object | Shows the schedule that dictates the execution of the job. There are two types of schedules: Cron and Interval. |

Author: Fixed

| `type` | String | Schedule type: Cron or Interval. |
| `start_time` | String | Time stamp start time associated with the Schedule. Interval schedule type only. |
| `interval` | Integer | Duration between job executions. Ex. minutes, hours, days. Interval schedule type only. |
| `unit` | String | Interval unit type. Ex. minutes, hours, days. Interval schedule type only. |
| `delay` | string | Fixed offset applied to this specific job execution. |
| `expression` | string | Cron schedule expression. Cron schedule type only. |
| `timezone` | string | Time zone associated with the Cron schedule. Cron schedule type only. |

> | `lock_duration` | integer | Length of the lock during job execution. |

Author: Changed the wording.

      "last_expected_execution_time": "2025-08-01T22:28:45.353111804Z",
      "next_expected_execution_time": "2025-08-01T22:29:45.353111804Z",
      "schedule": {
        "type": "interval",
        "start_time": "1970-07-23T00:27:45.353Z",
        "interval": 1,
        "unit": "Minutes",
        "delay": "none"
      },
      "lock_duration": 10,
      "jitter": "none"
    }
  ],
  "failures": [],
  "total_jobs": 2
}
```

## Response body fields

The following table lists all response body fields.

| Field | Data type | Description |
| :--- | :--- | :--- |
| `jobs` | Array | Contains all jobs reported by the Job Scheduler. |
| `job_type` | String | The plugin that scheduled the job. |
| `job_id` | String | The unique identifier of the job. |
| `index_name` | String | The index in which the job information is stored. |
| `name` | String | The job name. The name is not necessarily unique. |
| `descheduled` | Boolean | Indicates whether the job is scheduled to be executed (`false`) or is not scheduled (`true`) by the Job Scheduler. |
| `enabled` | Boolean | Indicates whether the job is active (`true`) or inactive (`false`), as defined by the plugin using the Job Scheduler. |

> | `enabled_time` | String | The time when the job was originally scheduled. |

Reviewer: ```suggestion
| `enabled_time` | String | The time at which the job was originally scheduled. |
```

> | `last_update_time` | String | The time the job was last updated. |

Reviewer: ```suggestion
| `last_update_time` | String | The time at which the job was last updated. |
```

> | `last_expected_exection_time` | String | The time when the job was most recently executed. |

Reviewer: ```suggestion
| `last_expected_exection_time` | String | The time at which the job was most recently executed. |
```

> | `next_expected_execution_time` | String | The time when the job is expected to be executed next. |

Reviewer: ```suggestion
| `next_expected_execution_time` | String | The time at which the job is expected to be executed next. |
```

| `schedule` | Map | The job's execution schedule. Can define a [Cron](#cron-schedule) or [interval](#interval-schedule) schedule. |
| `schedule.type` | String | The schedule type. Valid values are `cron` and `interval`. |

> | `lock_duration` | Integer | The maximum amount of time (in seconds) a job can remain locked during execution.|

Reviewer: ```suggestion
| `lock_duration` | Integer | The maximum amount of time (in seconds) that a job can remain locked during execution.|
```

> | `jitter` | Double | A random delay to job execution times to prevent simultaneous runs across the system.|

Reviewer: ```suggestion
| `jitter` | Double | A random delay applied to job execution times to prevent simultaneous runs across the system.|
```
